wikipedia_connector/wiki.py

```diff
@@ -99,48 +82,77 @@ def get_clean_text(self, text):
+    parser.add_argument('-f','--topics', action='store', dest='topics', default="table,chalk,starcraft,predator", required=True, help="""wikipedia topics""")
+    parser.add_argument('--project_dir', action='store', dest='project_dir',
+                        help="""--- For inner use of cnvrg.io ---""")
+    parser.add_argument('--output_dir', action='store', dest='output_dir',
+                        help="""--- For inner use of cnvrg.io ---""")
+    parser.add_argument(        "--local_dir",
+        action="store",
+        dest="local_dir",
+        required=False,
+        default=cnvrg_workdir,
+        help="""--- The path to save the dataset file to ---""",)
+    
+    args = parser.parse_args()
+    topic = args.topics
+    file_dir = args.local_dir
+    if topic.endswith(".csv"):
+        topics = pd.read_csv(topic)
+        flag_0 = "dataframe"
+    else:
+        topics = topic.split(",")
+        flag_0 = "list"
+
+
+    try:
+        if flag_0 == "dataframe":
+            input_list = []
+            list2 = []
+            disambiguation = 0
+            if(topics.shape[0]>0):
+                for i in range(topics.shape[0]):
+                    abc1 = WikiPage(str(topics.iloc[i, 0]))
+                    abc11 = abc1.get_wiki_page(str(topics.iloc[i, 0]))[0]
+                    input_list.append(abc11)
+                    if(abc1.get_wiki_page(str(topics.iloc[i, 0]))[1] == '1'):
+                        disambiguation = disambiguation + 1
+                    list2.append(str(topics[i]))
+            else:
+                for j in range(len(topics.columns.values)):
+                    abc2 = WikiPage(str(topics.columns.values[j]))
+                    abc22 = abc2.get_wiki_page(str(topics.columns.values[j]))[0]
+                    input_list.append(abc22)
+                    if(abc2.get_wiki_page(str(topics.columns.values[j]))[1] == '1'):
+                        disambiguation = disambiguation + 1
+                    list2.append(str(topics[i]))
+            #lst2 = ['x'] * len(input_list)
+            df1 = pd.DataFrame(list(zip(input_list, list2)), columns =['text', 'title'])
+            print('There were ' + str(disambiguation) + ' ambigious values in the input list')
+            output_summaries_file = "wiki_output.csv"
+            df1.to_csv(file_dir+"/{}".format(output_summaries_file), index=False)
         else:
-            for j in range(len(topics.columns.values)):
-                abc2 = WikiPage(str(topics.columns.values[j]))
-                abc22 = abc2.get_wiki_page(str(topics.columns.values[j]))[0]
-                input_list.append(abc22)
-                if(abc2.get_wiki_page(str(topics.columns.values[j]))[1] == '1'):
+            input_list = []
+            list2 = []
+            disambiguation = 0
+            for i in range(len(topics)):
+                abc1 = WikiPage(str(topics[i]))
+                abc11 = abc1.get_wiki_page(str(topics[i]))[0]
+                input_list.append(abc11)
```

Comment:
Not sure why no check on topic shape in this case? Probably not applicable?

Let's dicuss

Reply:
@orionsBeltWest Yea, it is not applicable - it is only applicable when input is given in .csv format